src/OrbitGl/include/OrbitGl/CaptureViewElement.h

```diff
@@ -164,13 +164,29 @@ class CaptureViewElement : public Pickable, public AccessibleInterfaceProvider {
   [[nodiscard]] virtual EventResult OnMouseEnter();
   [[nodiscard]] virtual EventResult OnMouseLeave();
 
+  // If TRUE, a new, uniquely named render group will be created for this element, and all of its
+  // content and that of its children will use this group for rendering. The content will also be
+  // restricted to the extents of the element automatically. By default, this returns false -
+  // override the method in a sub-class to change the behavior. Keep in mind that his has
+  // performance implications as it potentially adds a large number of render groups.
+  [[nodiscard]] virtual bool RequestSeparateRenderGroup() const { return false; }
+
+  [[nodiscard]] uint32_t GetUid() const { return uid_; }
+
  private:
   bool is_mouse_over_ = false;
+  uint32_t uid_;
+
+  std::string previous_batcher_render_group_name_{BatchRenderGroupId::kGlobalGroup};
+  std::string previous_text_render_group_name_{BatchRenderGroupId::kGlobalGroup};
```

Comment:
As far as I understand it the purpose of these variables is to communicate between `PreRender` and `PostRender` and are not used anywhere else.

`PreRender` and `PostRender` are only called from `Draw` and I don't see why the state of these variables needs to persist between `Draw` calls. Am I missing something?

In case I don't, may I suggest to create a struct that transfers the state between `PreRender` and `PostRender` and lives on the stack in the `Draw` stack frame? Something like this:

```cpp
struct RenderGroups {
  std::string batcher_render_group_name;
  std::string text_render_group_name;
};

[[nodiscard]] RenderGroups PreRender(PrimitiveAssembler&, TextRenderer&);
void PostRender(RenderGroups&&, PrimitiveAssembler&, TextRenderer&);
```

Reply:
Interesting... yes, it's only used like you described. That's a great suggestion, thanks.